content/en/tutorials/working-with-grpc-client.md

```diff
@@ -33,101 +33,130 @@ import (
-The `GetPendingCommands` command supports various filter options. If all of them are **unset**, all pending commands of all devices user is authorized to use are returned. 
+### Get pending commands
+
+The `GetPendingCommands` command supports various filter options. If all of them are **unset**, all pending commands of all devices user is authorized to use are returned.
 
 **Example usages of filter options:**
+
 - to retrieve pending commands of resources identified by their hrefs use `GetPendingCommandsRequest.resource_id_filter` where combinations `deviceID` and `href` is required in format `deviceID/href`
 - to retrieve pending commands of certain devices use `GetPendingCommandsRequest.device_id_filter` where ids of these devices need to be set
 - to retrieve pending commands of a specific type of resources use `GetPendingCommandsRequest.type_filter`
 - to retrieve pending commands of certain commands use `GetPendingCommandsRequest.command_filter`
 
 To return certain pending commands of binary switch resources hosted by devices with ids `deviceID1` and `deviceID2`, following options shall be set: `GetPendingCommandsRequest.device_id_filter("[deviceID1, deviceID2]") && RetrievePendingCommandsRequest.type_filter([oic.r.switch.binary])`.
 
-## Get devices metadata
+### Get devices metadata
+
 The `GetDevicesMetadata` command supports various filter options. If all of them are **unset**, all metadata of all devices user is authorized to use are returned. Metadata contains information about connection status(ONLINE/OFFLINE) and shadow synchronization(ENABLED, DISABLED)
 
 **Example usages of filter options:**
+
 - to retrieve metadata of certain devices use `GetDevicesMetadataRequest.device_id_filter` where ids of these devices need to be set
 - to retrieve metadata a specific type of devices use `GetDevicesMetadataRequest.type_filter`
 
-## Update Device Metadata
+### Update Device Metadata
+
 The `UpdateDeviceMetadata` command requests enable/disable shadow synchronization on the device.
 
-# Contracts
+### Get events
+
+The `GetEvents` command returns events that occurred on a resource. The command supports several filtering options. If all of them are **unset** all events from all resources belonging to given user devices are returned.
+
+**Example usages of filter options:**
+
+- to retrieve events for all user resources that occurred after given time use `GetEventsRequest.timestamp_filter` and set it to Unix timestamp in nanoseconds that represents given time.
+- to retrieve events for resources from selected user device after given time combine `GetEventsRequest.device_id_filter` and `GetEventsRequest.timestamp_filter`. The `GetEventsRequest.device_id_filter` parameter can be repeated multiple times.
+- to retrieve events for selected user resources after given time combine `GetEventsRequest.resource_id_filter` and `GetEventsRequest.timestamp_filter`. The `GetEventsRequest.resource_id_filter` parameter can be repeated multiple times.
+
```

Comment:
pls add there example how to get events from multiple devices (`deviceID1` and `deviceID2`) from certain timestamp.
similar as
```
To return certain pending commands of binary switch resources hosted by devices with ids `deviceID1` and `deviceID2`, following options shall be set: `GetPendingCommandsRequest.device_id_filter("[deviceID1, deviceID2]") && RetrievePendingCommandsRequest.type_filter([oic.r.switch.binary])`.
```